src/librustc/ty/query/keys.rs

```diff
@@ -12,7 +13,9 @@ use rustc_span::{Span, DUMMY_SP};
 
 /// The `Key` trait controls what types can legally be used as the key
 /// for a query.
-pub(super) trait Key {
+pub trait Key {
+    type CacheSelector;
```

Comment:
```suggestion
    type CacheSelector = DefaultCacheSelector;
```

Reply:
@Zoxc Is this actionable?

Reply:
It's unstable and I ran into some issues trying to use it.